src/pages/groupComparison/GroupComparisonStore.ts

```diff
@@ -3,29 +3,64 @@ import {SampleGroup} from "./GroupComparisonUtils";
 import {remoteData} from "../../shared/api/remoteData";
 import ListIndexedMap from "../../shared/lib/ListIndexedMap";
 import {Sample, SampleIdentifier} from "../../shared/api/generated/CBioPortalAPI";
+import {action, computed, observable} from "mobx";
 
 export default class GroupComparisonStore {
     constructor(
         private resultsViewPageStore:ResultsViewPageStore
     ) {
     }
 
-    public readonly sampleGroups = remoteData({
-        await:()=>[
-            this.resultsViewPageStore.samples,
+    @observable private _sampleGroup1Id:string;
+    @observable private _sampleGroup2Id:string;
+
+    @action public setSampleGroups(id1:string, id2:string) {
+        this._sampleGroup1Id = id1;
+        this._sampleGroup2Id = id2;
+    }
+
+    @computed get selectedGroupIds() {
```

Comment:
there is a possibility for selected groups to be more than 2

Reply:
according to @jjgao we are only doing pairwise comparisons for enrichments. so i guess the group 1 group 2 is enrichment specific

Reply:
true, for enrichments we require two. but there is a possibility that user can select more than groups(if present)